imgui-core/src/main/kotlin/imgui/static/dockNode.kt

```diff
@@ -1117,15 +1115,12 @@ fun dockNodePreviewDockRender(hostWindow: Window, hostNode: DockNode?, rootPaylo
         val tabBarRect = Rect()
         dockNodeCalcTabBarLayout(data.futureNode, null, tabBarRect, null)
         val tabPos = Vec2(tabBarRect.min)
-        hostNode!!.tabBar.let {
-            if (it != null)
-                tabPos.x += when {
-                    // We don't use OffsetNewTab because when using non-persistent-order tab bar it is incremented with each Tab submission.
-                    !hostNode.isHiddenTabBar && !hostNode.isNoTabBar -> it.offsetMax + style.itemInnerSpacing.x
-                    else -> style.itemInnerSpacing.x + tabItemCalcSize(hostNode.windows[0].name, hostNode.windows[0].hasCloseButton).x
-                }
-            else if (hostWindow.flags hasnt Wf._DockNodeHost)
-                tabPos.x += style.itemInnerSpacing.x + tabItemCalcSize(hostWindow.name, hostWindow.hasCloseButton).x // Account for slight offset which will be added when changing from title bar to tab bar
+        hostNode?.tabBar?.let {
+            tabPos.x += when {
+                // We don't use OffsetNewTab because when using non-persistent-order tab bar it is incremented with each Tab submission.
+                !hostNode.isHiddenTabBar && !hostNode.isNoTabBar -> it.offsetMax + style.itemInnerSpacing.x
+                else -> style.itemInnerSpacing.x + tabItemCalcSize(hostNode.windows[0].name, hostNode.windows[0].hasCloseButton).x
+            } // Account for slight offset which will be added when changing from title bar to tab bar
```

Comment:
nice catch on nullability, but where is the last `else if` branch?

Reply:
This went away when I let intellij optimise the null check - I'll revert it!

Reply:
Bloody Idea

Recently I tend to avoid refractoring from the IDE right because of this